src/Orleans.Core/Configuration/CollectionAgeLimitAttribute.cs

```diff
@@ -0,0 +1,29 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Orleans.Configuration
+{
+
+    [AttributeUsage(AttributeTargets.Class, AllowMultiple = false)]
+    public class CollectionAgeLimitAttribute : Attribute
+    {
+        private double minutes = CollectionAgeLimitConstants.DefaultCollectionAgeLimitInMinutes;
+
+        public double Minutes
+        {
+            get
+            {
+                return minutes;
+            }
+            set
+            {
+                if (value <= 0d)
+                {
+                    throw new ArgumentOutOfRangeException("Collection Age Limit must be a positive number.");
```

Comment:
We should remove this, or move it to the getter. The reason is that if the setter fails then the user will see a TypeLoadException at runtime and they are hard to debug.

An alternative approach is to create an implementation of `IConfigurationValidator` which receives `IOptions<GrainCollectionOptions>` in its constructor and then checks every one of the values in its `ValidateConfiguration` method. The benefit of doing it that way is that it will catch all cases where the user enters an invalid value (including when they configure `GrainCollectionOptions` in code) and the user will see the exception on startup and it's consistent with how the rest of the configuration is validated.

Reply:
This approach looks very nice. Let me see if I can add this capability.